core/src/common.cpp

```diff
@@ -327,76 +327,90 @@ namespace irods::http
 		boost::trim(bearer_token);
 		logging::debug("{}: Bearer token: [{}]", __func__, bearer_token);
 
+		const auto& config = irods::http::globals::configuration();
+
+		bool token_expired = false;
+		authenticated_client_info client_info_copy;
+
 		// Verify the bearer token is known to the server. If not, return an error.
-		auto mapped_value{irods::http::process_stash::find(bearer_token)};
-		if (!mapped_value.has_value()) {
-			const auto& config = irods::http::globals::configuration();
-
-			// It's possible that the admin didn't include the OIDC configuration stanza.
-			// This use-case is allowed, therefore we check for the OIDC configuration before
-			// attempting to access it. Without this logic, the server would crash.
-			static const auto oidc_conf_exists{
-				config.contains(nlohmann::json::json_pointer{"/http_server/authentication/openid_connect"})};
-			if (oidc_conf_exists) {
-				nlohmann::json json_res;
-
-				// Try parsing token as JWT Access Token
-				try {
-					auto token{jwt::decode<jwt::traits::nlohmann_json>(bearer_token)};
-					auto possible_json_res{openid::validate_using_local_validation(token)};
-
-					if (possible_json_res) {
-						json_res = *possible_json_res;
-					}
-				}
-				// Parsing of the token failed, this is not a JWT access token
-				catch (const std::exception& e) {
-					logging::debug("{}: {}", __func__, e.what());
-				}
+		const auto found_token = irods::http::process_stash::visit(bearer_token, [&](boost::any& _v) {
+			auto& client_info = boost::any_cast<authenticated_client_info&>(_v);
+			const auto now = std::chrono::steady_clock::now();
 
-				// Use introspection endpoint if it exists and local validation fails
-				static const auto introspection_endpoint_exists{
-					irods::http::globals::oidc_endpoint_configuration().contains("introspection_endpoint")};
-				if (json_res.empty() && introspection_endpoint_exists) {
-					auto possible_json_res{openid::validate_using_introspection_endpoint(bearer_token)};
-					if (possible_json_res) {
-						json_res = *possible_json_res;
-					}
-				}
+			if (now >= client_info.expires_at) {
+				token_expired = true;
+				return;
+			}
 
-				if (json_res.empty()) {
-					logging::error("{}: Could not find bearer token matching [{}].", __func__, bearer_token);
-					return {.response = fail(status_type::unauthorized)};
-				}
+			// Extend the lifetime of the bearer token.
+			static const auto token_lifetime =
+				config.at(nlohmann::json::json_pointer{"/http_server/authentication/basic/timeout_in_seconds"})
+					.get<int>();
+			client_info.expires_at = now + std::chrono::seconds{token_lifetime};
```

Comment:
One thing I'm thinking about is whether the `token_lifetime` value should have a dedicated config property.

We can start with this and add a new property later if people want/need that. I think that's the right approach. We're still likely a long way away from 1.0 status, so there's time.